syntax = "proto3";

package code_assistance;

service AgentService {
  rpc ExecuteTask (UserQuery) returns (stream AgentResponse);
  rpc SyncFile (FileUpdate) returns (SyncStatus);
}

message UserQuery {
  string project_id = 1;
  string prompt = 2;
  string session_id = 3;
}

message AgentResponse {
  string phase = 1; // "retrieving", "compressing", "reasoning", "final"
  string payload = 2;
  repeated string citations = 3;
}

// ðŸš€ ADDED: Definitions required by the Service
message FileUpdate {
  string project_id = 1;
  string file_path = 2;
  string content = 3;
}

message SyncStatus {
  bool success = 1;
  string message = 2;
}